central/ui/mux.go

```diff
@@ -16,8 +16,17 @@ func Mux() http.Handler {
 		http.ServeFile(w, r, "/ui/service-worker.js")
 	})
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("X-Frame-Options", "sameorigin")
-		w.Header().Set("Content-Security-Policy", "frame-ancestors 'self'")
+		headers := map[string]string{
+			"Cache-control":             "no-store, no-cache",
+			"Content-Security-Policy":   "frame-ancestors 'self'",
+			"Strict-Transport-Security": "max-age=31536000; includeSubDomains",
+			"X-Content-Type-Options":    "nosniff",
+			"X-Frame-Options":           "sameorigin",
+			"X-XSS-Protection":          "1; mode=block",
```

Comment:
@rhybrillou It would be nice if you could add a comment to code with a reference to why you added each of the headers you did.

One minor question for the @stackrox/ui-dep-updaters :  the `Cache-control: no-store` setting, I think, will cause the user's browser to re-download all the frontend app's JS and CSS files every time they close a browser tab and then visit the app again. I'm not sure if the trade-off in efficiency vs security is necessary for static assets.

@rhybrillou , Is it possible to only add `no-store` to API responses, and not static assets, like JS, CSS, and images?

Reply:
My tests also show that the `cache-control` headers do not appear on responses from `/static/` endpoints. Mozilla re-used JS and CSS files from its cache.
`cache-control` headers were added to the `/api/graphql`, `/v1/` and `/v2/` endpoints.